app/controllers/api/v1/facts_controller.rb

```diff
@@ -83,13 +81,17 @@ def set_fact
     render json: { error: "Fact not found" }, status: :not_found unless @fact
   end
   
-  def check_access 
-    if member.present?
-      if member.user_id != current_user.id
-        render json: { message: "The current user is not authorized for that data."}, status: :unauthorized
-      end
-    else
-      render json: { error: "Member not found" }, status: :not_found
+  def set_member
+    
+    @member = Member.find_by(id: params[:member_id])
+    
+  end
+
+  def check_access
+    if (@member.user_id != current_user.id) 
```

Comment:
Hmm, looks like the tests don't handle the case where the member with the given ID doesn't exist, but that's more of a problem with Swagger (or the instructions we provide?) rather than with the work you did, so don't worry about it. Just know that with `find_by`, you may get a `nil` value, which of course doesn't have a `user_id` so `check_access` would break. We can test this with the Rails console:

```ruby
irb(main):001:0>  @member = Member.find_by(id: -1)
  Member Load (0.7ms)  SELECT "members".* FROM "members" WHERE "members"."id" = ? LIMIT ?  [["id", -1], ["LIMIT", 1]]
=> nil                                                                         
irb(main):002:1* if (@member.user_id != current_user.id)
irb(main):003:1*   puts "No error!?"
irb(main):004:0> end
(irb):2:in `<main>': undefined method `user_id' for nil:NilClass (NoMethodError)
                                                                               
if (@member.user_id != current_user.id)                                        
           ^^^^^^^^
```

The `nil?` check that you had previously would make sense to have after a `find_by` if there's a reason to not just have a generic `rescue_from ActiveRecord::RecordNotFound` handler.